README.md

```diff
@@ -65,13 +94,83 @@ The Marsha syntax is meant to be:
+                        Code generation with no correction stages run
+  -a ATTEMPTS, --attempts ATTEMPTS
+  -n N_PARALLEL_EXECUTIONS, --n-parallel-executions N_PARALLEL_EXECUTIONS
+  --exclude-main-helper
+                        Skips addition of helper code for running as a script
+  -s, --stats           Save stats and write them to a file
+```
+
+* `-d` adds a significant amount of debug information to the screen. Probably not useful if you're not working on Marsha itself.
+* `-q` runs only the initial code generation phase without any of the corrective feedback stages. This is significantly cheaper, but more likely to generate code that doesn't quite work. This could be useful if you're using Marsha like Github Copilot or directly asking for code from ChatGPT, but with the Marsha syntax providing some more structure to produce a better result than you might if simply given a blank screen to write into.
+* `-a` The number of times marsha should attempt to compile your program, defaulting to just once. If set to more than 1, on a failure it will try again. For some trickier programs this might improve the ability to get working code at the cost of more LLM calls.
+* `-n` The number of parallel LLM threads of "thought" to pursue per attempt. This defaults to 3. When a path succeeds, all of the other paths are cancelled.
+* `-s` Save the stats that are printed by default to a file, instead. Probably not useful if you're not working on Marhsa itself.
+* `--exclude-main-helper` Turns off the automatically generated code to make using your compiled Marsha code from the CLI easier, which is included by default.
+
+## Using compiled Marsha code
+
+By default, Marsha appends logic to the generated Python code to make usage simpler, allowing you to invoke it from the CLI and potentially start a REST server.
+
+```sh
+$ python -m duckduckgo --help
+usage: duckduckgo.py [-h] [-c {BeautifulSoup,duckduckgo}] [-j] [-t] [-i] [-f INFILE] [-o OUTFILE] [-s SERVE] [params ...]
+
+Marsha-generated CLI options
+
+positional arguments:
+  params                Arguments to be provided to the function being run. Optimistically converted to simple python types by default, and left as strings if not possible
+
+options:
+  -h, --help            show this help message and exit
+  -c {BeautifulSoup,duckduckgo}, --func {BeautifulSoup,duckduckgo}
+                        Specifies the function to call. Defaults to the last defined function
+  -j, --force-json      Forces arguments, files, or stdin to be parsed as JSON
+  -t, --force-text      Forces arguments, files, or stdin to be parsed as raw text
+  -i, --stdin           Ignores CLI parameters in favor of stdin (as a single parameter)
+  -f INFILE, --infile INFILE
+                        Ignores CLI parameters in favor of reading the specified file (as a single parameter)
+  -o OUTFILE, --outfile OUTFILE
+                        Saves the result to a file instead of stdout
+  -s SERVE, --serve SERVE
```

Comment:
should not be the port specified here? is there a default port?

Reply:
That's what the "SERVE" text is. I don't know how to get Python's `argparse` to instead display `--serve PORT`.

There is no default, if you don't specify the port it will error at you and tell you to provide one.